### PR TITLE
Add prefix_list property to bgp_neighbor_af

### PR DESCRIPTION
**Pull request by chrisvanheuveln**
This is a boilerplate prop modeled on filter-list and route-map.

rubocop/beaker tests are clean.

### CODE REVIEW

**Comment by chrisvanheuveln** [comment]
This PR should wait for my node_utils PR to be merged first.

File: lib/puppet/type/cisco_bgp_neighbor_af.rb [hunk 1]
```diff
@@ -384,6 +386,24 @@ def self.title_patterns
     newvalues(:true, :false, :default)
   end
 
+  newproperty(:prefix_list_in) do
+    desc 'prefix-list in state. Valid values are a string defining the name ' \
+         "of the prefix-list or 'default'."
+    munge do |value|
+      value = :default if value == 'default'
```
Comment (hunner) [inline lib/puppet/type/cisco_bgp_neighbor_af.rb]: Given that the next line casts `value` to a string, this line isn't needed is it?

Comment (chrisvanheuveln) [inline lib/puppet/type/cisco_bgp_neighbor_af.rb]: It is needed -- it's the value.to_s that's wrong and that's breaking our default handling. Unfortunately we used this same code in a couple of places so I've got more to fix than just this one. New diff coming up...

**Comment by mikewiebe** [comment]
:+1:

**Pull request merged**